File: apps/mobile/src/services/inventoryMatch.js
export const QUOTE_DISCLAIMER = '推荐车源仅用于人工报价前评估，不等同于正式报价。';

function formatPrice(value, currency = 'USD') {
  if (value == null) {
    return '价格待确认';
  }
  return `${currency} ${Number(value).toLocaleString('en-US')}`;
}

export function buildInventoryMatchView(match = {}) {
  const validUntil = match.validUntil || match.valid_until || '';
  const title = `${match.brand || 'Unknown'} ${match.model || 'Unknown'}${match.year ? ` ${match.year}` : ''}`;
  const exportReady = match.exportReady ?? match.export_ready;
  return {
    matchId: match.matchId || match.match_id,
    title,
    reason: match.recommendationReason || match.recommendation_reason || 'Unknown',
    priceText: formatPrice(match.quotedPrice ?? match.quoted_price, match.currency || 'USD'),
    exportLabel: exportReady ? '可出口' : '不可出口',
    expiryLabel: validUntil ? `有效至 ${validUntil.slice(0, 10)}` : '有效期 Unknown',
    riskTips: match.riskTips || match.risk_tips || [],
    priorityRecommendable: Boolean(match.priorityRecommendable ?? match.priority_recommendable),
    requiresComplianceReview: Boolean(match.requiresComplianceReview ?? match.requires_compliance_review),
    quoteDisclaimer: QUOTE_DISCLAIMER,
  };
}

export function filterRecommendableMatches(matches = []) {
  return matches.filter((match) => Boolean(match.priorityRecommendable ?? match.priority_recommendable));
}

export function buildMatchDecisionPayload({ decision, owner, note }) {
  return {
    decision,
    owner,
    note,
    formal_quote_allowed: false,
  };
}

File: apps/api/app/schemas/inventory_match.py
from uuid import UUID

from pydantic import BaseModel, Field


class InventoryMatchRequest(BaseModel):
    vehicle_type: str | None = None
    min_year: int | None = None
    max_price: float | None = None
    requires_compliance_review: bool = False


class InventoryMatchItemResponse(BaseModel):
    match_id: UUID
    inventory_item_id: UUID
    inventory_external_id: str | None
    brand: str
    model: str
    year: int | None
    vehicle_type: str | None
    condition_summary: str | None
    quoted_price: float | None
    currency: str
    export_ready: bool
    valid_until: str | None
    priority_recommendable: bool
    recommendation_reason: str
    risk_tips: list[str]
    requires_compliance_review: bool


class InventoryMatchListResponse(BaseModel):
    customer_id: UUID
    quote_disclaimer: str
    items: list[InventoryMatchItemResponse]


class InventoryMatchDecisionRequest(BaseModel):
    decision: str = Field(pattern="^(advance_quote|not_match)$")
    owner: str = Field(min_length=1)
    note: str | None = None


class InventoryMatchDecisionResponse(BaseModel):
    match_id: UUID
    decision: str
    owner: str
    note: str | None
    formal_quote_allowed: bool
    next_gate: str
